.gitignore

```diff
@@ -2,3 +2,4 @@ node_modules/
 lib/
 coverage/
 .nyc_output/
+src/dbconfig.json
```

Comment:
Why did you commit a file and then add the same file to the .gitignore?

Reply:
I committed the file to have an example as to how the file should look like. When you pull the project (if I remember correctly) it will be added along with your project, but any changes won't be committed when you commit them remotely, so you can edit in the username/password/etc and not have it be published for everyone to see.

Reply:
I'd give [this](https://github.com/motdotla/dotenv) a look.  Add config vars (such as `PORT` and the ones you're proposing) in a `.env` file and then do something like this with `index.js`:

```javascript
import http from "http";
import dotenv from "dotenv";
import app from "./app";

dotenv.config();

const PORT= process.env.PORT;

const server = http.createServer(app);
server.listen(PORT, err => {
	if (err) {
		throw err;
	}
	console.log(`Server listening on port ${PORT}`);
});
```

Then all your config vars are stored on `process.env`, which is the same place they'll be found in production anyways

Reply:
Ahh, looks a lot like configparser in python. Will do that.

Reply:
And `.env` would be in `.gitignore`, like you had above

Reply:
The `dotenv` README has a lot of great info in it

Reply:
Yeah I was just gonna pull it all from JSON files, but dotenv seems a lot less hack-y.